src/WebClient.ts

```diff
@@ -144,46 +144,37 @@ export class WebClient extends EventEmitter {
             agent: this.agentConfig,
           }, this.tlsConfig),
         )
+          .then((response: got.Response<string>) => {
+            const result = this.buildResult(response);
+            // log warnings in response metadata
+            if (result.response_metadata !== undefined && result.response_metadata.warnings !== undefined) {
+              result.response_metadata.warnings.forEach(this.logger.warn);
+            }
+            return result;
+          })
           .catch((error: got.GotError) => {
             // Wrap errors in this packages own error types (abstract the implementation details' types)
             if (error.name === 'RequestError') {
               throw requestErrorWithOriginal(error);
             } else if (error.name === 'ReadError') {
               throw readErrorWithOriginal(error);
             } else if (error.name === 'HTTPError') {
+              // Special case: retry if 429
+              const result = this.buildResult(error.response);
+              if (error.statusCode === 429) {
+                const retryAfterMs = result.retryAfter !== undefined ? result.retryAfter : (60 * 1000);
+                this.emit('rate_limited', retryAfterMs / 1000);
+                if (this.retryConfig.retries !== 0) {
```

Comment:
ah-hah. i think you found an issue that actually resides within the tests. the case `should allow rate limit triggered retries to be turned off` was noted assuming that's functionality we intend to support. but some time after that note, it was made clear that the `retryConfig` has nothing to do with the rate-limit control built into this object. see: https://github.com/slackapi/node-slack-sdk/issues/451

specifically, i think we need to remove this condition and think about that problem more holistically. right now you check a very specific `retries` property of the `retryConfig`, but its actually a complex object that's meant to be interpreted by a more sophisticated library like `retry`. long story short: we need to solve for that test case in a different way, and we should probably remove its implementation for now.

we can start work on resolving #451, but i think that should happen in a separate PR.